src/rustup.rs

```diff
@@ -101,7 +101,21 @@ pub fn component_is_installed(component: &str, toolchain: &str, verbose: bool) -
         .any(|l| l.starts_with(component) && l.contains("installed")))
 }
 
-pub fn rustc_version(toolchain_path: &Path) -> Result<Option<(Version, String)>> {
+fn rustc_channel(version: &str) -> Channel {
+    // the channel starts with `"{major}.{minor}.{patch}"`, and can
+    // end also have -beta.{revision} or -nightly after version.
+    if version.contains("dev") {
+        Channel::Dev
+    } else if version.contains("nightly") {
+        Channel::Nightly
+    } else if version.contains("beta") {
+        Channel::Beta
+    } else {
+        Channel::Stable
+    }
+}
```

Comment:
this is how rustc_version does this

```suggestion
fn rustc_channel(version: &Version) -> Channel {
    let channel = match version.pre.split('.').next().unwrap() {
        "" => Channel::Stable,
        "dev" => Channel::Dev,
        "beta" => Channel::Beta,
        "nightly" => Channel::Nightly,
        x => return Err(Error::UnknownPreReleaseTag(x.to_owned())),
    };
}
```

I think I prefer that approach. (atleast using match instead of a if else chain)